src/Components/ProductGrid/ProductGrid.js

```diff
@@ -1,20 +1,43 @@
 /* eslint-disable react/prop-types */
-import React, { useContext } from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
 import ProductCard from '../ProductCard';
-import Products from '../../MockData/featured-products.json';
-import { FilterContext } from '../../Pages/ProductsPage/ProductsPage';
 import './styles.scss';
 
 const ProductGrid = (props) => {
-  const { showProductsButton, title, activeFilter } = props;
-  let products = Products.results;
+  const { showProductsButton, title, activeFilter, productData, setSearchParams, searchParams } =
+    props;
+  const [disabledPrev, setDisabledPrev] = useState(false);
+  const [disabledNext, setDisabledNext] = useState(false);
 
-  const { filterState } = useContext(FilterContext);
+  if (!productData) {
+    return (
+      // <Loading/>
+      <h3>Loading ...</h3>
+    );
+  }
+
+  let filteredProducts = productData.results || [];
+
+  function prevPage() {
+    let current = Number(searchParams.get('page'));
+    let q = searchParams.get('q') || '';
+    if (current == 1) setDisabledPrev(true);
+    else {
+      // setSearchParams({ page: current - 1 });
+      setSearchParams({ page: current - 1, q: q });
+      setDisabledPrev(false);
+    }
```

Comment:
Yo aquí le pondría un early return para quitar el else
```suggestion
    if (current == 1) return setDisabledPrev(true);
    setSearchParams({ page: current - 1, q: q });
    setDisabledPrev(false);
```
algo que se me ocurre podría cambiar y que también serviría para el de next page es buscar como ponder el `let current = Number(searchParams.get('page'));` para que te sirva en un useEffect y puedas hacer los disable ahi adentro, probablemente tendrías que cambiar mas cosas pero podría ser de utilidad